go.mod

```diff
@@ -17,41 +17,41 @@ require (
 	github.com/LeanerCloud/ec2-instances-info v0.0.0-20230905092627-1725cb4f820e
 	github.com/Masterminds/semver/v3 v3.2.1
 	github.com/Masterminds/sprig/v3 v3.2.3
-	github.com/aliyun/alibaba-cloud-sdk-go v1.62.619
+	github.com/aliyun/alibaba-cloud-sdk-go v1.62.654
 	github.com/aws/aws-sdk-go-v2 v1.24.1
-	github.com/aws/aws-sdk-go-v2/config v1.25.8
-	github.com/aws/aws-sdk-go-v2/credentials v1.16.6
-	github.com/aws/aws-sdk-go-v2/service/ec2 v1.137.3
+	github.com/aws/aws-sdk-go-v2/config v1.26.3
+	github.com/aws/aws-sdk-go-v2/credentials v1.16.14
+	github.com/aws/aws-sdk-go-v2/service/ec2 v1.142.1
 	github.com/aws/aws-sdk-go-v2/service/eks v1.34.2
 	github.com/aws/aws-sdk-go-v2/service/iam v1.27.5
 	github.com/aws/aws-sdk-go-v2/service/s3 v1.48.0
-	github.com/aws/aws-sdk-go-v2/service/sts v1.25.6
+	github.com/aws/aws-sdk-go-v2/service/sts v1.26.7
 	github.com/aws/smithy-go v1.19.0
 	github.com/cert-manager/cert-manager v1.13.1
 	github.com/cilium/cilium v1.13.7
-	github.com/digitalocean/godo v1.106.0
+	github.com/digitalocean/godo v1.107.0
 	github.com/distribution/distribution/v3 v3.0.0-20230629214736-bac7f02e02a1
 	github.com/envoyproxy/go-control-plane v0.11.1
 	github.com/go-git/go-git/v5 v5.11.0
 	github.com/go-logr/zapr v1.3.0
 	github.com/go-test/deep v1.1.0
 	github.com/gobuffalo/flect v1.0.2
 	github.com/google/go-containerregistry v0.16.1
-	github.com/google/uuid v1.4.0
-	github.com/gophercloud/gophercloud v1.7.0
+	github.com/google/uuid v1.5.0
+	github.com/gophercloud/gophercloud v1.8.0
 	github.com/hetznercloud/hcloud-go v1.52.0
 	github.com/imdario/mergo v0.3.16
 	github.com/jackpal/gateway v1.0.10
 	github.com/kubermatic/grafanasdk v0.9.13
-	github.com/kubermatic/machine-controller v1.58.1-0.20231206121500-4eb097a46352
+	github.com/kubermatic/machine-controller v1.58.2
```

Comment:
why bump this to a patch release and not the latest commit that is in `pkg/resources/machinecontroller/deployment.go`?

Reply:
Why not ?

Reply:
Because we usually keep Go library and component image in sync so that API types match.

Reply:
It's not a blocker to this PR; I was just curious.

Reply:
I see, no worries at all as that kinda makes sense. Will use the commit hash in the go mod file as well :)